Assets/HoloToolkit/Input/Scripts/Utilities/AttachToController.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity.InputModule;
+using UnityEngine;
+#if UNITY_2017_2_OR_NEWER
+using UnityEngine.XR.WSA.Input;
+#else
+using UnityEngine.VR.WSA.Input;
+#endif
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Waits for a controller to be instantiated, then attaches itself to a specified element
+    /// </summary>
+    public class AttachToController : MonoBehaviour
+    {
+        public InteractionSourceHandedness Handedness { get { return handedness; } }
+
+        public MotionControllerInfo.ControllerElementEnum Element { get { return element; } }
+
+        [Header("AttachToController Elements")]
+        [SerializeField]
+        protected InteractionSourceHandedness handedness = InteractionSourceHandedness.Left;
+
+        [SerializeField]
+        protected MotionControllerInfo.ControllerElementEnum element = MotionControllerInfo.ControllerElementEnum.PointingPose;
+
+        public bool SetChildrenInactiveWhenDetached = true;
+
+        [SerializeField]
+        protected Vector3 positionOffset = Vector3.zero;
+
+        [SerializeField]
+        protected Vector3 rotationOffset = Vector3.zero;
+
+        [SerializeField]
+        protected Vector3 scale = Vector3.one;
+
+        [SerializeField]
+        protected bool setScaleOnAttach = false;
+
+        public bool IsAttached { get; private set; }
+
+        private Transform elementTransform;
+        public Transform ElementTransform { get; private set; }
+
+        protected MotionControllerInfo controller;
+
+        protected virtual void OnAttachToController() { }
+        protected virtual void OnDetachFromController() { }
+
+        protected virtual void OnEnable()
+        {
+            SetChildrenActive(false);
```

Comment:
Should this be a `GameObject` extension method?

Reply:
I'll defer to @keveleigh on this one since it may be a special case.